packages/edit-site/src/components/layout/animation.js

```diff
@@ -1,125 +1,207 @@
+			window.matchMedia( '(prefers-reduced-motion: reduce)' ).matches // They prefer not.
+		) {
 			return;
 		}
 
-		// We disable the animation if the user has a preference for reduced
-		// motion.
-		const disableAnimation = window.matchMedia(
-			'(prefers-reduced-motion: reduce)'
-		).matches;
+		const animationConfig = flipBox(
+			{ from, to: ref.current.getBoundingClientRect() },
+			{ duration: ANIMATION_DURATION, easing: quintInOut }
+		);
 
-		if ( disableAnimation ) {
-			return;
+		const control = animate( ref.current, animationConfig );
+		return () => control.cancel();
+	}, [ from ] );
+
+	return ref;
+}
+
+export default useMovingAnimation;
+
+/** @param {number} t */
+const linear = ( t ) => t;
+
+/** @param {number} t */
+const quintInOut = ( t ) =>
+	( t *= 2 ) < 1
+		? 0.5 * t * t * t * t * t
+		: 0.5 * ( ( t -= 2 ) * t * t * t * t + 2 );
+
+/**
+ * @param {string} style
+ */
+function propertyToCamelCase( style ) {
+	const parts = style.split( '-' );
+	if ( parts.length === 1 ) {
+		return parts[ 0 ];
+	}
+	return (
+		parts[ 0 ] +
+		parts
+			.slice( 1 )
+			.map(
+				/** @param {any} word */ ( word ) =>
+					word[ 0 ].toUpperCase() + word.slice( 1 )
+			)
+			.join( '' )
+	);
+}
+
+/**
+ * @param {string} css
+ */
+function asKeyframes( css ) {
+	// eslint-disable-next-line jsdoc/no-undefined-types
+	/** @type {Keyframe} */
+	const keyframe = {};
+	const parts = css.split( ';' );
+	for ( const part of parts ) {
+		const [ property, value ] = part.split( ':' );
+		if ( ! property || value === undefined ) {
+			break;
 		}
 
-		const controller = new Controller( {
-			x: 0,
-			y: 0,
-			width: prevRect.width,
-			height: prevRect.height,
-			config: {
-				duration: ANIMATION_DURATION,
-				easing: easings.easeInOutQuint,
-			},
-			onChange( { value } ) {
-				if ( ! ref.current ) {
-					return;
-				}
-				let { x, y, width, height } = value;
-				x = Math.round( x );
-				y = Math.round( y );
-				width = Math.round( width );
-				height = Math.round( height );
-				const finishedMoving = x === 0 && y === 0;
-				ref.current.style.transformOrigin = 'center center';
-				ref.current.style.transform = finishedMoving
-					? null // Set to `null` to explicitly remove the transform.
-					: `translate3d(${ x }px,${ y }px,0)`;
-				ref.current.style.width = finishedMoving
-					? null
-					: `${ width }px`;
-				ref.current.style.height = finishedMoving
-					? null
-					: `${ height }px`;
-			},
-		} );
+		keyframe[ propertyToCamelCase( property.trim() ) ] = value.trim();
+	}
+	return keyframe;
+}
+
+/**
+ * Animates an element, according to the provided configuration.
+ * @param {Element}                  element
+ * @param {AnimationConfig}          config
+ * @param {number}                   target   The target progression — `1` is end, `0` is beginning.
+ * @param {(() => void) | undefined} callback
+ */
+const animate = ( element, config, target = 1, callback ) => {
```

Comment:
Is this some kind of generic utility you wrote on your own? Do we expect this to be reused elsewhere (block animation maybe as I just copied that one and adapted it to create this file)

Reply:
I pared down [Svelte’s internal `animate`](https://github.com/sveltejs/svelte/blob/84ad208f2e986fa549683a12e7f999c3d31b127f/packages/svelte/src/internal/client/dom/elements/transitions.js#L268). It would serve for reuse, the block animation could pass a different config (one that operates only on `transfrom` which should make that one run on the GPU). I’d thought we’d want to try that if this worked well in this instance.